lib/llm/src/kv_router.rs

```diff
@@ -141,3 +133,32 @@ impl KvRouter {
         Ok(worker_id)
     }
 }
+
+#[async_trait]
+impl AsyncEngine<SingleIn<RouterRequest>, ManyOut<Annotated<RouterResponse>>, Error> for KvRouter {
+    async fn generate(
+        &self,
+        request: SingleIn<RouterRequest>,
+    ) -> Result<ManyOut<Annotated<RouterResponse>>> {
+        let (request, ctx) = request.into_parts();
+        let isl_tokens = request.tokens.len();
+        let block_size = self.block_size;
+
+        // Compute the block hashes in a blocking task
+        let local_block_hashes: Vec<LocalBlockHash> = tokio::task::spawn_blocking(move || {
+            Tokens::compute_block_hash(&request.tokens, block_size)
+                .into_iter()
+                .map(LocalBlockHash)
+                .collect()
+        })
```

Comment:
Is it expensive enough to justify a background thread?

Reply:
benchmarking coming, but for now yes. when we get the criterion bits setup i'll add a branching logic.  it's not spawning a thread but is scheduling to a thread. I think the cost is on the order of 1-2us.